packages/jest-mock/__typetests__/mock-functions.test.ts

```diff
@@ -66,7 +66,8 @@ expectType<Mock<(e: any) => never>>(
 );
 expectError(fn('moduleName'));
 
-const mockFn = fn((a: string, b?: number) => true);
```

Comment:
Arrow functions cannot have `this` type parameters, so I changed this to use a declared type. This seems fine because there is no runtime execution.